### ISSUE TITLE
Dynamic icon code block not working

### ISSUE THREAD

**Issue by akanshSirohi** [DanielPantle/strapi-plugin-react-icons#22]
I have been trying to use the following code block provided in README.md but it seems to be not working. It shows error that the module `react-icons/all` not found with the latest version of react-icons. Please update the snippet or just provide the solution to fix this issue.

```ts
import * as ReactIcons from "react-icons/all";

interface IIconComponent {
  icon: string;
  size?: number;
}
type IReactIcon = keyof typeof ReactIcons;

const IconComponent: React.FC<IIconComponent> = ({ icon, size }) => {
  const DynamicIconComponent = ReactIcons[icon as IReactIcon];

  if (undefined === DynamicIconComponent) return <></>;

  return <DynamicIconComponent size={size} />;
};
```

**Comment by akanshSirohi** [comment]
After trying several different methods, I have successfully created a working component. I am posting it here for anyone like me who finds it useful. It works with latest react-icons.

```jsx
import dynamic from 'next/dynamic';

const iconComponents = {
  Ai: () => import('react-icons/ai'),
  Bs: () => import('react-icons/bs'),
  Bi: () => import('react-icons/bi'),
  Ci: () => import('react-icons/ci'),
  Di: () => import('react-icons/di'),
  Fi: () => import('react-icons/fi'),
  Fc: () => import('react-icons/fc'),
  Fa: () => import('react-icons/fa6'),
  Gi: () => import('react-icons/gi'),
  Go: () => import('react-icons/go'),
  Gr: () => import('react-icons/gr'),
  Hi: () => import('react-icons/hi2'),
  Im: () => import('react-icons/im'),
  Lia: () => import('react-icons/lia'),
  Io: () => import('react-icons/io5'),
  Lu: () => import('react-icons/lu'),
  Md: () => import('react-icons/md'),
  Pi: () => import('react-icons/pi'),
  Rx: () => import('react-icons/rx'),
  Ri: () => import('react-icons/ri'),
  Si: () => import('react-icons/si'),
  Sl: () => import('react-icons/sl'),
  Tb: () => import('react-icons/tb'),
  Tfi: () => import('react-icons/tfi'),
  Ti: () => import('react-icons/ti'),
  Vsc: () => import('react-icons/vsc'),
  Wi: () => import('react-icons/wi'),
  Cg: () => import('react-icons/cg'),
};

export default function ReactIcon(props) {
  const { icon, size, color } = props;

  for(let i = 3; i >= 2; i--) {
    const iconComponentImport = iconComponents[icon.substring(0, i)];
    if (iconComponentImport) {
      const DynamicIcon = dynamic(() =>
        iconComponentImport().then((icons) => icons[icon])
      );
      return <DynamicIcon size={size} color={color} />;
    }
  }

  return <></>;
}
```

How to use it:-

```jsx
import ReactIcon from "./reactIcon";
<ReactIcon icon="FaAngellist" size={20} color="white" />
```

**Comment by martinnaste** [comment]
A way to get around the missing/changed imports is to install the same version of "react-icons" that the plugin uses - 4.8.0. Then update the iconComponents to something like this. Otherwise works as intended. Thanks!

`const iconComponents: iconComponent = {
  Ai: () => import("react-icons/ai"),
  Bs: () => import("react-icons/bs"),
  Bi: () => import("react-icons/bi"),
  Ci: () => import("react-icons/ci"),
  Di: () => import("react-icons/di"),
  Fi: () => import("react-icons/fi"),
  Fc: () => import("react-icons/fc"),
  Fa: () => import("react-icons/fa"),
  Gi: () => import("react-icons/gi"),
  Go: () => import("react-icons/go"),
  Gr: () => import("react-icons/gr"),
  Hi: () => import("react-icons/hi2"),
  Im: () => import("react-icons/im"),
  Lib: () => import("react-icons/lib"),
  Io: () => import("react-icons/io5"),
  Md: () => import("react-icons/md"),
  Rx: () => import("react-icons/rx"),
  Ri: () => import("react-icons/ri"),
  Si: () => import("react-icons/si"),
  Sl: () => import("react-icons/sl"),
  Tb: () => import("react-icons/tb"),
  Tfi: () => import("react-icons/tfi"),
  Ti: () => import("react-icons/ti"),
  Vsc: () => import("react-icons/vsc"),
  Wi: () => import("react-icons/wi"),
  Cg: () => import("react-icons/cg"),
};`

**Comment by SashaYursa** [comment]
How i can use it with ts?

**Comment by DanielPantle** [comment]
Resolved with #26 (v0.3.0)
